languages/python/sqlalchemy-oso/tests/test_post_relationship.py

```diff
@@ -784,6 +792,15 @@ def test_partial_isa_with_path(session, oso, tag_nested_many_many_test_fixture):
 
     assert len(posts) == 5
 
+    tags = session.query(Tag).filter(authorize_model(oso, user, "read", session, Tag))
```

Comment:
Against main, this test fails with: 

```
field_name = 'is_public', value = True, operator = 'Unify', session = <sqlalchemy.orm.session.Session object at 0x7fbd382e2e20>, model = <class 'tests.models.Post'>

    def emit_compare(field_name, value, operator, session, model):
        """Emit a comparison operation comparing the value of ``field_name`` on ``model`` to ``value``."""
        assert not isinstance(value, Variable), "value is a variable"
>       property = getattr(model, field_name)
E       AttributeError: type object 'Post' has no attribute 'is_public'

sqlalchemy_oso/partial.py:352: AttributeError
```

Since all the `check` rules will pass type checking, and it incorrectly attempts to check `is_public` on a post.

Reply:
This is what this PR is addressing. Previously `post` from `post in tag.posts` would be "untyped", so it would match all of the `check` rules. With this PR, we will check `tag.posts matches Post`, etc. So only the right rule gets applied.